### PR TITLE
Teach python SDK not to send data back to the server if it has not ch…

### PR DESCRIPTION
**Pull request by DanielCollins**
…anged

since it was received.

### CODE REVIEW

File: python/util/rights.py [hunk 1]
```diff
@@ -29,7 +29,7 @@ def to_dict(self):
                 return_codes.append(code)
         return return_codes
 
-    def from_json(self, right_codes):
+    def from_json(self, right_codes, makes_dirty=True):
```
Comment (s3341458) [inline python/util/rights.py]: looks like a param which has never been actually used.

Comment (DanielCollins) [reply python/util/rights.py]: it's not used in this definition, but you want the prototypes of `from_json` to be the same.

Comment (s3341458) [reply python/util/rights.py]: got it :+1:

File: python/tests/test_entity.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    assert j.client.name == 'turtle'
+    output = j.serialise(exclude_old=True)[0]
+    assert 'client' not in output
+    assert 'quantity' not in output
+    # now we touch the users name
+    j.client.name = 'elephant'
+    # although j.client was not touched directly we need it in the output now
+    output = j.serialise(exclude_old=True)[0]
+    assert output['client']['name'] == 'elephant'
+    assert 'quantity' not in output
```
Comment (s3341458) [inline python/tests/test_entity.py]: the behaviour of from_json when makes_dirty is true, is not covered.
the behaviour of from_json on a recurisive attribute which is a list of object is not covered in the testcase.

**Comment by s3341458** [comment]
every time I read a @DanielCollins which is more than 50 lines of code, I will learn something.

**Pull request merged**